### PR TITLE
Feature: new security section, links

### PR DESCRIPTION
**Pull request by waterim**
### Details

Added new Security section to readme, added links to all add/remove member places.

### Fixed Issues

$ https://github.com/Expensify/App/issues/33575
PROPOSAL: https://github.com/Expensify/App/issues/33575#issuecomment-1898824403

### Tests

Verify the rules added in `#Security` section of README.md are working as expected

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as tests

### QA Steps

None

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [x] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>Web</summary>

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -394,6 +395,74 @@ This application is built with the following principles.
 
 ----
 
+# Security
+Updated rules for managing all types of chats (add/remove people)
+
+1. ### DM
```
[the remark this answers is not in the document]
Comment (waterim) [reply README.md:401]: Its not possible on markdown to make ordered list bold unfortunately

**Comment by waterim** [comment]
@aimane-chnaif 

> iv. `Workspace rooms` seems not correct to me.
> 
> permission
> Member should be divided into default and invited
> 
> Current behavior: default member: show Members option so they can Invite/Remove others invited member: show Invite option so they can only Invite others

Now everyone can delete everyone in the workspace room as discussed [here](https://github.com/Expensify/App/issues/33575#issuecomment-1915458928), its not a workspace chat which is creating automatically, its a workspace room.
image

**Comment by waterim** [comment]
@aimane-chnaif Ahh, okay, I see, you are right, good catch, updating it.

**Comment by waterim** [comment]
@aimane-chnaif I think we need to add a possibility for Guests to leave the workspace room as well

**Comment by waterim** [comment]
@aimane-chnaif Okay, updated workspace room is correct.
Everyone can leave the workspace room right now(members and guests).
image

**Comment by waterim** [comment]
@aimane-chnaif hey, do you want me to update something additionally? 
Or what is the blocker on the review?

File: README.md [hunk 2]
```diff
@@ -395,6 +396,100 @@ This application is built with the following principles.
...
+
+
+1. ### DM
+    |  | Member
+    | :---: | :---: 
+    | **Invite** | ❌ 
+    | **Remove** | ❌ 
+    | **Leave**  | ❌ 
+    | **Can be removed**  | ❌
+- **DM the only ever have 2 participants**
```
Comment (MonilBhavsar) [inline README.md:410]: A summary would be good
We can also remove bold style (from all descriptions)
```suggestion
- DM always has two participants. None of the participant can leave or be removed from the DM. Also no additional member can be invited to the chat.
```

File: README.md [hunk 3]
```diff
@@ -395,6 +396,100 @@ This application is built with the following principles.
...
+        | **Invite** | ✅ |  ❌ |  ✅ | ❌
+        | **Remove** | ✅ |  ❌ |  ✅ | ❌
+        | **Leave**  | ❌ |  ✅ |  ❌ | ✅
+        | **Can be removed**  | ❌ |  ✅ | ✅ | ✅
+
+        - **Creator can't leave or be removed from their own workspace**
+        - **Admins can't leave from the workspace**
+        - **Admins can remove other workspace admins, as well as workspace members, and invited guests**
+        - **Creator can remove other workspace admins, as well as workspace members, and invited guests**
+        - **Members and Auditoes can't invite or remove from the workspace they were invited to**
```
Comment (MonilBhavsar) [inline README.md:425]:
```suggestion
        - **Members and Auditors cannot invite or remove anyone from the workspace**
```

File: README.md [hunk 4]
```diff
@@ -395,6 +396,100 @@ This application is built with the following principles.
...
+
+    2. #### Workspace #announce room
+        |   |  Member(Employee/User) | Admin |  Auditor?
+        | :---: | :---:  |  :---: | :---: 
+        | **Invite** | ❌ |  ❌ |  ❌
+        | **Remove** | ❌ |  ❌ |  ❌
+        | **Leave**  | ❌ |  ❌ |  ❌
+        | **Can be removed**  | ❌ |  ❌ |  ❌ |
+
+       - **All members can't leave or be removed from the #announce room**
```
Comment (MonilBhavsar) [inline README.md:435]:
```suggestion
       - **No one can leave or be removed from the #announce room**
```

File: README.md [hunk 5]
```diff
@@ -395,6 +396,100 @@ This application is built with the following principles.
...
+    4. #### Workspace rooms
+        |   |  Creator | Member | Guest(outside of the workspace)
+        | :---: | :---:  |  :---: | :---:
+        | **Invite** | ✅ | ✅ | ✅
+        | **Remove** | ✅ | ✅ | ❌
+        | **Leave**  | ✅ | ✅ | ✅
+        | **Can be removed**  | ✅ | ✅ | ✅
+
+        - **Everyone can be removed/can leave from the rooms(creator as well)**
+        - **Only guests are not able to remove from the room**
```
Comment (MonilBhavsar) [inline README.md:456]:
```suggestion
        - **Guests are not able to remove anyone from the room**
```

File: README.md [hunk 6]
```diff
@@ -395,6 +396,100 @@ This application is built with the following principles.
...
+    
+    4. #### Workspace rooms
+        |   |  Creator | Member | Guest(outside of the workspace)
+        | :---: | :---:  |  :---: | :---:
+        | **Invite** | ✅ | ✅ | ✅
+        | **Remove** | ✅ | ✅ | ❌
+        | **Leave**  | ✅ | ✅ | ✅
+        | **Can be removed**  | ✅ | ✅ | ✅
+
+        - **Everyone can be removed/can leave from the rooms(creator as well)**
```
Comment (MonilBhavsar) [inline README.md:455]:
```suggestion
        - **Everyone can be removed/can leave from the room including creator**
```

**Review by aimane-chnaif — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -395,6 +396,120 @@ This application is built with the following principles.
...
+- **Nobody can leave or be removed from something they were automatically added to. For example:**
+
+    - DM members can't leave or be removed from their DMs
+    - Members can't leave or be removed from their own workspace chats
+    - Admins can't leave or be removed from workspace chats
+    - Members can't leave or be removed from the #announce room
+    - Admins can't leave or be removed from #admins
+    - Domain members can't leave or be removed from their domain chat
+    - Report submitters can't leave or be removed from their reports (eg, if they are the report.accountID)
+    - Report managers can't leave or be removed from their reports (eg, if they are the report.managerID)
```
Comment (MonilBhavsar) [inline README.md:411]: Let's remove this. Not really useful here
```suggestion
    - Report submitters can't leave or be removed from their reports
    - Report managers can't leave or be removed from their reports
```

**Comment by waterim** [comment]
@aimane-chnaif @quinthar approved updated security section: https://expensify.slack.com/archives/C0671JDRKQW/p1707794023284829?thread_ts=1703547187.140519&cid=C0671JDRKQW

File: README.md [hunk 8]
```diff
@@ -395,6 +396,120 @@ This application is built with the following principles.
...
+    - Group owners cannot be removed from their groups - they need to transfer ownership first
+- **Excepting the above, admins can remove anyone. For example:**
+    - Group admins can remove other group admins, as well as group members
+    - Workspace admins can remove other workspace admins, as well as workspace members, and invited guests
+- **Excepting the above, members can remove guests. For example:**
+    - Workspace members can remove non-workspace guests.
+- **Excepting the above, anybody can remove themselves from any object**
+
+
+
```
Comment (MonilBhavsar) [inline README.md:421]: Let's remove these extra line also

**Review by MonilBhavsar — APPROVED** [review]

**Pull request merged**